pkg/model/insight.proto

```diff
@@ -56,3 +56,55 @@ enum InsightApplicationCountLabelKey {
   KIND = 0;
   ACTIVE_STATUS = 1;
 }
+
+// time is treated as unix time second
+
+message InsightDeploymentSubset {
+  int64 created_at = 14 [(validate.rules).int64.gte = 0];
+  int64 updated_at = 15 [(validate.rules).int64.gte = 0];
+}
+
+message InsightDailyDeployment {
+  // `date` must be normalized like xxxx/xx/xx:00:00:00;
+  int64 date = 1 [(validate.rules).int64.gte = 0];
+
+  // Daily InsightDeploymentSubset sorted by created_at ASC.
+  repeated InsightDeploymentSubset daily_deployments = 4;
+
+  // Unix time when the insight daily deployment is created.
+  int64 created_at = 14 [(validate.rules).int64.gte = 0];
+  // Unix time of the last time when the insight daily deployment is updated.
+  int64 updated_at = 15 [(validate.rules).int64.gte = 0];
+}
+
+// InsightDeploymentChunk represents a chunk that is used by insight
+message InsightDeploymentChunk {
+  InsightChunkDateRange date_range = 1;
+
```

Comment:
```proto
enum InsightDeploymentVersion {
   // Comment
   V0 = 0;
}

message InsightDeploymentChunk {
  enum InsightDeploymentVersion version = 1;
  int64 from = 2 [(validate.rules).int64.gte = 0];
  int64 to = 3 [(validate.rules).int64.gte = 0];

  repeated InsightDeployment deployments = 4;
}

message InsightDeployment {
   string id = 1;
   string app_id = 2;
   map<string,string> labels = 3;

   int64 started_at = 4;
   int64 completed_at = 5;
   int64 rollback_started_at = 6;
   model.Deployment_Status complete_status = 7;
}

message InsightDeploymentChunkMetadata {
   message Chunk {
      int64 from = 1 [(validate.rules).int64.gte = 0];
      int64 to = 2 [(validate.rules).int64.gte = 0];
      string name = 3;
      int64 size = 4;
      int64 count = 5;
  }

   repeated Chunk chunks = 1;

   int64 created_at = 14;
   int64 updated_at = 15;
}
```

Reply:
save meta first. and then save chunkdata